README.rst

```diff
@@ -0,0 +1,320 @@
+        "sub": {
+            "in_queue": "/home/user/queues/queue"
+        },
+        "queues": {
+            "/home/user/queues/queue": {
+                "path": "/home/user/queues/queue",
+                "max_reader_num": 1024,
+                "max_db_num": 1024,
+                "max_db_size_bytes": 34359738368,
+                "subscribers": ["sub"],
+                "high-water-mark": {
+                    "MSG_TIMEOUT_SECS": 600,
+                    "MAX_MSGS_NUM": 10000,
+                    "HWM_LMDB_SIZE_BYTES": 1000000,
+                    "strategy": "prune_first"
+                }
+            }
+        }
+    }
+
+Control
+^^^^^^^
+
+Initialize all queues
+"""""""""""""""""""""
+
+.. code-block:: python
+
+    import persipubsub.control
+
+    persipubsub.control.initialize_all_dbs(config_pth="/home/user/config.json")
+
+Prune all dangling messages
+"""""""""""""""""""""""""""
+
+.. code-block:: python
+
+    import persipubsub.control
+
+    persipubsub.control.prune_dangling_messages(
+        config_pth="/home/user/config.json")
+
+Clear all messages
+""""""""""""""""""
+
+.. code-block:: python
+
+    import persipubsub.control
+
+    persipubsub.control.clear_all_subs(config_pth="/home/user/config.json")
+
+
+Runtime
+-------
+During runtime only publisher and subscriber are needed. Control can be
+optionally be used for pruning although the queues prune itself on a regular
+basis.
+
+Publisher
+^^^^^^^^^
+
+Initialization
+""""""""""""""
+
+Assuming that all queues were initialized during deployment the publisher can
+be initialized as following.
+
+.. code-block:: python
+
+    import persipubsub.publisher
+
+    pub = persipubsub.publisher.Pub()
+    pub.init(pub_id="pub", config_pth="/home/user/config.json")
+
+Send a message
+""""""""""""""
+
+.. code-block:: python
+
+    msg = "Hello there!".encode('utf-8')
+    pub.send(msg=msg)
+
+    # subscribers have now a message in the queue.
+
+Send many messages at once
+""""""""""""""""""""""""""
+
+.. code-block:: python
+
+    msgs = ["What's up?".encode('utf-8'),
+            "Do you like the README?".encode('utf-8')]
+    pub.send_many(msgs=msgs)
+
+    # Subscribers have now both messages in the queue not necessary in the list
+    # order.
+
+Subscriber
+^^^^^^^^^^
+
+Initialization
+""""""""""""""
+
+Assuming that all queues were initialized during deployment the publisher can
+be initialized as following.
+
+.. code-block:: python
+
+    import persipubsub.subscriber
+
+    sub = persipubsub.subscriber.Sub()
+    sub.init(sub_id="sub", config_pth="/home/user/config.json")
+
+Receive a message
+"""""""""""""""""
+
+.. code-block:: python
+
+    # one message in queue
+    with sub.receive() as msg:
+        # do something with the message
+        print(msg)  # b'Hello there!'
+
+    # sub queue is now empty
+
+Catch up with latest message
+""""""""""""""""""""""""""""
+
+In case your subscriber is slow, the queue is growing and you want to want to
```

Comment:
In case when the subscriber's loop is spinning slower then the publisher's, there is a possibility receive the latest message and discard the others ones.